src/server/pfs/drive/driver.go

```diff
@@ -821,6 +821,49 @@ func (d *driver) DeleteAll(shards map[uint64]bool) error {
 	return nil
 }
 
+func (d *driver) ArchiveAll(shards map[uint64]bool) error {
+	repoInfos, err := d.ListRepo(nil, shards)
+	if err != nil {
+		return err
+	}
+	// We want to make sure we archive repos after archiving their provenance.
+	// The reason is that if a pipeline see's missing output Commits for a set of input Commits, it will trigger the pipeline, creating new output commits. This is a race.
+
+	// Provenance has a nice invariant:
+	// A in Provenance(B) => len(Provenance(A)) < len(Provenance(B))
+	// Since Provenance is transitive
+	// Thus when we sort by length we guarantee that A will have a lower index than B
+	// Because we want to make sure we delete the higher indexes first we traverse in reverse
+	sort.Sort(byProvenance(repoInfos))
+	var repos []*pfs.Repo
+	for _, repoInfo := range repoInfos {
+		// We only need to archive repos with no provenance
+		// i.e. all raw repos
+		if len(repoInfo.Provenance) > 0 {
+			break
+		}
+		repos = append(repos, repoInfo.Repo)
+	}
+	commitInfos, err := d.listCommit(
+		repos,
+		pfs.CommitType_COMMIT_TYPE_NONE,
+		nil,
+		nil,
+		pfs.CommitStatus_ALL,
```

Comment:
This should be `pfs.CommitStatus_NORMAL`. There's no need to archive cancelled commits since they already don't show up in `ListFile` and there's no reason to archive commits that are already archived.